backend/controllers/appointments/addCourtAppointment.js

```diff
@@ -143,11 +130,58 @@ const addCourtAppointmentsFactory = (existingEventsService, elite2Api, logError)
 
       const preAppointmentLocationsValid = preAppointmentRequired === 'yes' ? preLocations.length : true
       const postAppointmentLocationsValid = postAppointmentRequired === 'yes' ? postLocations.length : true
+      const noAvailabilityForGivenTime = Boolean(
+        !mainLocations.length || !preAppointmentLocationsValid || !postAppointmentLocationsValid
+      )
+      if (noAvailabilityForGivenTime) {
+        const start =
+          preAppointmentRequired === 'yes'
+            ? startTime.subtract(25, 'minutes').format(DATE_TIME_FORMAT_SPEC)
+            : startTime.format(DATE_TIME_FORMAT_SPEC)
+
+        const end =
+          postAppointmentRequired === 'yes'
+            ? endTime.add(25, 'minutes').format(DATE_TIME_FORMAT_SPEC)
+            : endTime.format(DATE_TIME_FORMAT_SPEC)
+
+        const availableRooms = await availableSlotsService.getAvailableRooms(res.locals, {
+          startTime: start,
+          endTime: end,
+          agencyId,
+        })
 
-      if (!mainLocations.length || !preAppointmentLocationsValid || !postAppointmentLocationsValid) {
-        return res.send('No rooms available')
+        const atLeastTwoRoomsNeeded = Boolean(preAppointmentRequired === 'yes' || postAppointmentRequired === 'yes')
+        const minRooms = atLeastTwoRoomsNeeded ? 2 : 1
```

Comment:
I don't really understand the domain - why do you need 2 rooms if doing a pre or post appointment? Suggests both rooms would be used at the same time.

Reply:
You need at least 2 because of the following:

- Pre and post meeting can use the same room
- Pre and post meeting can't use the same location as the main one

Reply:
Also worth noting that the pre and post appointments are 25 min each and the available slots service brings back locations that are available from the start time of the _pre_ and the end time of the _post_. With that it shouldn't be possible for the pre to overlap with the main appointment.